### ISSUE TITLE
influxql: time duration literals should require whitespace after

### ISSUE THREAD

**Issue by mark-rushakoff** [influxdata/influxdb#7002]
Working from 11d141ae1fadfcac2a36e23d4aa38b36271a5daf, the following influxql statements are parsed even though they look like they're missing whitespace between the duration literal and a following keyword:

```
> CREATE DATABASE r WITH DURATION 1dREPLICATION 1SHARD DURATION 0mNAME m
> show retention policies on r
name	duration	shardGroupDuration	replicaN	default
m	24h0m0s		1h0m0s			1		true

> CREATE CONTINUOUS QUERY ""ON x RESAMPLE EVERY 0sFOR 2hBEGIN SELECT n(e)INTO "y"FROM "x"GROUP BY time(1s)END
> show continuous queries
name: x
-------
name	query
	CREATE CONTINUOUS QUERY  ON x RESAMPLE FOR 2h BEGIN SELECT n(e) INTO r.m.y FROM r.m.x GROUP BY time(1s) END
```

**Comment by bbczeuz** [comment]
Another issue with whitespacing in queries: InfluxDB should be tolerant with spacing:
This should be supported: ```SELECT max(value) FROM "series" WHERE time>now()-1d GROUP BY time(1h)```
--> No spaces between time, now() and -1d. Elements can still be separated by the operators.

**Comment by jsternberg** [comment]
This ends up getting fixed by #6597.

@bbczeuz I think I know what the issue is with that one. Lexers have a lot of trouble with that kind of number. Can you file an issue for that so I can take care of it separately? Thanks.
